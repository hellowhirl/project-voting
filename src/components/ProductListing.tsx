import { Product } from "../App";

type ProductProps = {
  product: Product;
  upvote: (id: number) => void;
};

export default function ProductListing({ product, upvote }: ProductProps) {
  return (
    <div className="product-listing">
      <div className="product-image">
        <img src={product.productImageUrl} />
      </div>
      <div className="product-details">
        <div className="votes-container">
          <img
            onClick={() => upvote(product.id)}
            alt={`arrow-icon-${product.id}`}
            src="./images/ui/arrow.png"
          />
          <div
            className="votes"
            role="vote-count"
            data-testid={`voteId-${product.id}`}
          >
            {product.votes}
          </div>
        </div>
        <a href={product.url}>
          <div className="title">{product.title}</div>
        </a>
        <div className="description">{product.description}</div>
        <div className="submitted-container">
          <p>Submitted by:</p>
          <div className="avatar-cropper">
            <img src={product.submitterAvatarUrl} />
          </div>
        </div>
      </div>
    </div>
  );
}
