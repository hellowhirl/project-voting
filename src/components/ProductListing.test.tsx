import React from "react";
import { render, screen } from "@testing-library/react";
import ProductListing from "./ProductListing";
import renderer from "react-test-renderer";

const fakeProduct = {
  id: 1,
  title: "Yellow Pail",
  description: "On-demand sand castle construction expertise.",
  url: "#",
  votes: 12,
  submitterAvatarUrl: "images/avatars/daniel.jpg",
  productImageUrl: "images/products/image-aqua.png",
};

test("renders product name", () => {
  render(<ProductListing product={fakeProduct} upvote={() => {}} />);
  const titleElement = screen.getByText(/Yellow Pail/i);
  expect(titleElement).toBeInTheDocument();
});

test("renders correct vote number", () => {
  render(<ProductListing product={fakeProduct} upvote={() => {}} />);
  const voteElement = screen.getByText(/12/i);
  expect(voteElement).toBeInTheDocument();
});

it("renders correctly", () => {
  const tree = renderer
    .create(<ProductListing product={fakeProduct} upvote={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
