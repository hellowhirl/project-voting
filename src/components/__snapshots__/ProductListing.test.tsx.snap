// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product-listing"
>
  <div
    className="product-image"
  >
    <img
      src="images/products/image-aqua.png"
    />
  </div>
  <div
    className="product-details"
  >
    <div
      className="votes-container"
    >
      <img
        alt="arrow-icon-1"
        onClick={[Function]}
        src="./images/ui/arrow.png"
      />
      <div
        className="votes"
        data-testid="voteId-1"
        role="vote-count"
      >
        12
      </div>
    </div>
    <a
      href="#"
    >
      <div
        className="title"
      >
        Yellow Pail
      </div>
    </a>
    <div
      className="description"
    >
      On-demand sand castle construction expertise.
    </div>
    <div
      className="submitted-container"
    >
      <p>
        Submitted by:
      </p>
      <div
        className="avatar-cropper"
      >
        <img
          src="images/avatars/daniel.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
